src/schema-parser/spectral-rule-v2.ts

```diff
@@ -0,0 +1,101 @@
+import { createRulesetFunction } from '@stoplight/spectral-core';
+import { aas2_0, aas2_1, aas2_2, aas2_3, aas2_4 } from '@stoplight/spectral-formats';
+
+import { validateSchema, getSchemaFormat, getDefaultSchemaFormat } from './index';
+import { createDetailedAsyncAPI } from '../utils';
+
+import type { RuleDefinition } from "@stoplight/spectral-core";
+import type { Parser } from '../parser';
+import type { ValidateSchemaInput } from './index';
+import type { SchemaValidateResult } from '../types';
+
+export function aas2schemaParserRule(parser: Parser): RuleDefinition {
+  return {
+    description: 'Custom schema must be correctly formatted from the point of view of the used format.',
+    formats: [aas2_0, aas2_1, aas2_2, aas2_3, aas2_4],
+    message: '{{error}}',
+    severity: 'error',
+    type: 'validation',
+    recommended: true,
+    given: [
+      // operations
+      '$.channels.*.[publish,subscribe].message',
```

Comment:
I just realized all those paths are tied to AsyncAPI Spec v2.x.x. How would we make it happen for v3.x.x ?

Reply:
By separate function, we can use different "formats" for rules in this same rulesets - one function for v2, another one for v3. Don't worry :)